NEWS.md

```diff
@@ -54,6 +54,7 @@
 * `any_is_na_linter()` is extended to catch the unusual usage `NA %in% x` (#2113, @MichaelChirico).
 * `make_linter_from_xpath()` errors up front when `lint_message` is missing (instead of delaying this error until the linter is used, #2541, @MichaelChirico).
 * `paste_linter()`  is extended to recommend using `paste()` instead of `paste0()` for simply aggregating a character vector with `collapse=`, i.e., when `sep=` is irrelevant (#1108, @MichaelChirico).
+* `scalar_in_linter` is now configurable to allow other `%in%` like operators to be linted. `%chin%` is no longer linted by default. (@F-Noelle)
```

Comment:
Since the default argument is `NULL` for this parameter, the second sentence is no longer needed here.

Reply:
The linter previously did lint the `%chin%` in its unconfigurable state. As this is no longer the case due to the default changing to NULL this is a notable change. If there is a better way to phrase that I am all ears.

Reply:
The larger point of the change you are making is allowing to configure this linter. `%chin%` was just _your_ use case, and the developers who are reading this NEWS item don't need to know that. For users of some other package, it would have been a different operator (say `%nin%`).

I personally don't use `{data.table}` much, so I wasn't even aware of this infix operator; as a result, this part of the NEWS item was confusing for me because I thought this was an operator in base that I didn't know about.

Reply:
So you might misunderstand something here. `%chin%` was never my use case. I know of `{data.tables}`, but never used it in any of my projects. I just through it would be easy to make this linter configurable to work with my use cases, hence the pull request.

Previously the linter in its unconfigurable form linted `%in%` and `%chin%`, nothing else. When I made the update I wanted to make it backwards compatible to its original use case, so I opted to make `%chin%` the default for the new parameter. 

@MichaelChirico then suggested to demote `%chin%` from its special status making this PR a breaking change, which I am completely on board with. But having a breaking change is NEWS worthy in my opinion, because people might have relied on the old behaviour aka `%chin%` being linted. 

As this is no longer the case I want to highlight that simple fact as part of the NEWS. Just because I don't rely on it doesn't mean that nobody else did. If there is a better way of phrasing this I am all ears.

Reply:
I have nothing more to add here over my previous reply. 

I will let @MichaelChirico decide the best phrasing for this news item, if he thinks it merits being phrased differently.

Reply:
If we do make a breaking change, the corresponding bullet should be in its own section to highlight the change more clearly.

Reply:
I agree with @F-Noelle about the NEWS item, it looks good to me. Maybe we can emphasize better that it's a {data.table} operator.

@IndrajeetPatil see here, I'm not quite sure your point:

https://github.com/r-lib/lintr/blob/fc2268fb4b8609ac16fdba0b112319008529e11c/R/scalar_in_linter.R#L35

 - On current `main`, `x %chin% 'a'` will lint
 - On this PR, `x %chin% 'a'` will only lint under `scalar_in_linter("%chin%")` or similar.